### PR TITLE
predict verbal perseverations early postop using random forest based on ANTS ratios 

### PR DESCRIPTION
**Pull request by matthieufaillot**
this code attempts to predicts verbal perseverations early postop using random forest based on ANTS ratios

### CODE REVIEW

File: first_script.py [hunk 1]
```diff
@@ -522,4 +522,81 @@ def make_spider (row, title, color):
...
+
+from sklearn.model_selection import train_test_split
+X_train, X_test, Y_train, Y_test = train_test_split (X, Y, test_size=0.2, random_state=0)
+
+#this normalisation step makes the performance drop
+#from sklearn.preprocessing import StandardScaler
+#sc = StandardScaler ()
+#X_train= sc.fit_transform (X_train)
+#X_test = sc.fit_transform (X_test)
+
```
Comment (bthirion) [inline first_script.py:655]: Indeed you should never normalize train and test independently. You can remove these lines.

File: first_script.py [hunk 2]
```diff
@@ -522,4 +522,81 @@ def make_spider (row, title, color):
...
+from sklearn.ensemble import RandomForestClassifier
+classifier=RandomForestClassifier (n_estimators=45, random_state=0)
+classifier.fit(X_train, Y_train)
+Y_pred= classifier.predict(X_test)
+
+from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
+print (confusion_matrix (Y_test,Y_pred))
+print (classification_report(Y_test,Y_pred))
+print (accuracy_score(Y_test,Y_pred))
+
```
Comment (bthirion) [inline first_script.py]: Here we really need to get the results from a cross validation with lots of folds.
```
rs = ShuffleSplit(n_splits=100, test_size=.2, random_state=0)

```
then measure the accuracy with `sklearn.model_selection.cross_val_score`

File: first_script.py [hunk 3]
```diff
@@ -522,4 +522,81 @@ def make_spider (row, title, color):
...
+importances = classifier.feature_importances_
```
Comment (bthirion) [inline first_script.py]: On the other hand, you can compute importances from the whole dataset, but given the small size they won't be reliable.

File: first_script.py [hunk 4]
```diff
@@ -522,4 +522,81 @@ def make_spider (row, title, color):
...
+# Plot the impurity-based feature importances of the forest
+plt.figure()
+plt.title("Feature importances")
+plt.bar(range(X.shape[1]), importances[indices],
+        color="r", yerr=std[indices], align="center")
+plt.xticks(range(X.shape[1]), X_column_values[indices],rotation=90,size=7)
+plt.xlim([-1, X.shape[1]])
+plt.show()
+
+#afficher les réseaux 4 et 12 de Yeo dans NiLearn
```
Comment (bthirion) [inline first_script.py]: why have you commented the following lines ?

Comment (matthieufaillot) [reply first_script.py]: I wanted to display Yeo 4th and 12th network since they seemed to be important for the classifier. It didn't work and I was planning on fixing it later. My second thought is that plotting Yeo's networks is not very helpful at this stage.

File: first_script.py (hunk 3 above)
Comment (matthieufaillot) [reply first_script.py]: ok I understand, I removed these lines as well

File: first_script.py (hunk 1 above)
Comment (matthieufaillot) [reply first_script.py:655]: Ok sorry I thought I was applying the same transformation to both, but obviously this was not the case...

File: first_script.py (hunk 2 above)
Comment (matthieufaillot) [reply first_script.py]: Ok I think I understand the point.
I modified the code accordingly. Mean accuracy is not  great ( around 0.5..)
Maybe I need to use another classifier, or to put other parameters as input (such as age, preop neuropsychological assessment etc...)

Comment (matthieufaillot) [reply first_script.py]: also maybe I need to do  as you suggested (line 54) :  to find a more sophisticated averaging scheme to deal with low values since those values are unreliable

**Comment by matthieufaillot** [comment]
I tried to find another method for averaging data ( for each subject add the different ratios computed in each HCP connectome), still some values are still very low and this does not improve the performance
I tried to transform data (first method and second method) with a sigmoid function but this does not improve the performance
I was thinking of performing PCA decomposition on the connectome data, before adding other variables (age, preop neuropsychological assessment)

**Comment by matthieufaillot** [comment]
applying pca doesn't improve prediction performance

**Comment by bthirion** [comment]
Thx ! I'm going to merge now.
